### DISCUSSION TITLE
Private Assistant user attached - bug when modifying ?

### DISCUSSION THREAD

**Discussion by djuillard** [danny-avila/LibreChat#4193] [Help Wanted]
Hi,

I wanted to "prepare" the config of an assistant on behalf of another user in the application, and then set it to "private" while changing the user-attachment to that new user (instead of me).

I have then tested the privateAssistants option in the endpoint of my librechat.yaml file and also checked how the application handled the link between the users and the assistant owner. I checked in the mongodb assistants collection, and I understood that such a link is made when the assistant is created, and in that case, there is object "user" in the newly created assistant, and the value of the user-key is the object_id of the user. 

Examples  : 
  {
    _id: ObjectId('66e836d68699fd15ce054edb'),
    assistant_id: 'asst_54wUqf850ITN5wLdn2OGOl9P',
    __v: 0,
    conversation_starters: [],
    createdAt: ISODate('2024-09-16T13:47:02.517Z'),
    updatedAt: ISODate('2024-09-16T13:49:18.988Z')
  },
  {
    _id: ObjectId('66e9238408395f2e6544d46c'),
    assistant_id: 'asst_nGrLGmgRxs3U42WBZIkqYBAA',
    __v: 0,
    conversation_starters: [],
    createdAt: ISODate('2024-09-17T06:36:52.335Z'),
    updatedAt: ISODate('2024-09-17T06:36:52.335Z'),
    user: ObjectId('669fc648eccae303d1447432')
  }

Here, 
- for the first assistant, there is no user object, because this assistant has directly been created into the assistant playground of the openai or azure openai platform. In that case, this has not been created by a librechat user, and therefore, if privateAssistants is set to "true", no one (except librechat ADMIN) will have access to this assistant through librechat. That seems logical.
- for the second assistant, there is one user object and the value is ObjectId('669fc648eccae303d1447432') corresponding to one of the user of librechat application. Indeed, this is the user who created the assistant directly from librechat through the assistant builder. And if we set privateAssistants to "true", that given user (+ librechat ADMIN) will have access to this assistant through librechat, but all other users won't be able to access it. Again, this is very logical.

My intention, was to change the user object and update it with the value of the user I am building the assistant for. But, when I update that value into mongoDB, it does not change anything in the application. The new user can't see the assistant, and I - as the assistant creator - still see and access the assistant whereas I "removed" my user value off the user object in the assistant.

How can you explain that ? Is there another way to proceed ?

Complementary question -> in case this is a bug and this can be fixed, do you think that we could pass an array instead of one single value for user so that we can "cheat" and make it like there are several user's owners.... which would end up to give access to one given assistant to multiple users (like team sharing).

Thanks a lot in advance,

**Comment by danny-avila** [comment]
There is metadata associated on the Assistant on OpenAI's side as well, but I'm not sure you can edit this using their playground, although i can see it on their network requests.

![image](https://github.com/user-attachments/assets/011a68b1-84f9-45c1-913f-310c6f014e56)

In short, you should create the assistant from LC to begin with.

I would ditch assistants for Agents when I release them this week, since sharing/private functionalities will be built in without dependence to some external service

https://github.com/danny-avila/LibreChat/issues/3607

**Comment by djuillard** [comment]
Thanks for your reply. I knew that there were some metadata on openai's side, but I could figure out whether the user id was included as well (and I did not think of checking network requests...)

Thanks -> I'll wait for agents then.
